packages/transaction-controller/src/TransactionController.ts

```diff
@@ -776,23 +776,48 @@ export class TransactionController extends BaseController<
             value: '0x0',
           };
 
-    const unsignedEthTx = this.prepareUnsignedEthTx(txParams);
+    const unsignedEthTx = this.prepareUnsignedEthTx(cancelTxParams);
 
     const signedTx = await this.sign(
       unsignedEthTx,
       transactionMeta.txParams.from,
     );
     await this.updateTransactionMetaRSV(transactionMeta, signedTx);
     const rawTx = bufferToHex(signedTx.serialize());
-    await query(this.ethQuery, 'sendRawTransaction', [rawTx]);
-    transactionMeta.estimatedBaseFee = estimatedBaseFee;
-    transactionMeta.status = TransactionStatus.cancelled;
+    const hash = await query(this.ethQuery, 'sendRawTransaction', [rawTx]);
+    const cancelTransactionMeta: TransactionMeta = {
+      ...transactionMeta,
```

Comment:
Since there is so many properties in the metadata, and potentially more in future, is it safer to define all the new properties explicitly in case we're pulling in values that aren't applicable?

Reply:
Make sense, in that case, I will only pull `chainId` and it's a complete `transaction`
Fixed in https://github.com/MetaMask/core/pull/1998/commits/ddc794a32aa14746ff95b08f613b89723ec222f1